src/borrow_tracker/mod.rs

```diff
@@ -0,0 +1,337 @@
+pub mod stacked;
+
+use std::cell::RefCell;
+use std::fmt;
+use std::num::NonZeroU64;
+
+use log::trace;
+use smallvec::SmallVec;
+
+use rustc_data_structures::fx::{FxHashMap, FxHashSet};
+use stacked::diagnostics::RetagCause;
+
+use crate::*;
+
+pub type CallId = NonZeroU64;
+
+/// Tracking pointer provenance
+#[derive(Copy, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BorTag(NonZeroU64);
+
+impl BorTag {
+    pub fn new(i: u64) -> Option<Self> {
+        NonZeroU64::new(i).map(BorTag)
+    }
+
+    pub fn get(&self) -> u64 {
+        self.0.get()
+    }
+
+    pub fn inner(&self) -> NonZeroU64 {
+        self.0
+    }
+
+    pub fn succ(self) -> Option<Self> {
+        self.try_map(|n| n.checked_add(1))
+    }
+
+    // The default to be used when SB is disabled
+    pub fn one() -> Self {
```

Comment:
```suggestion
    pub fn default() -> Self {
```
No need to expose its value.